### PR TITLE
[BISERVER-12392] Data Source is not removed completely if it is deleted from PIR/PA "Select Data Source" wizard page

### PR DESCRIPTION
**Pull request by IvanNikolaychuk**
- When deleting logical model, delete it's both (reporting and analysis) representations.
- Delete domain if it has no logical models.
- Tests written
- Checkstyle applied

@mdamour1976, @e-cuellar, review it please.

### CODE REVIEW

**Comment by IvanNikolaychuk** [comment]
@mdamour1976, @e-cuellar, review it please.

**Comment by IvanNikolaychuk** [comment]
@mdamour1976, @e-cuellar, review it please.

File: src/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/DSWDatasourceServiceImpl.java [hunk 1]
```diff
@@ -185,7 +185,18 @@ public boolean deleteLogicalModel( String domainId, String modelName ) throws Da
...
+      getMetadataDomainRepository().removeModel( domainId, logicalModel.getId() );
+
+      if ( logicalModelRep != null && !logicalModelRep.getId().equals( logicalModel.getId() ) ) {
+        getMetadataDomainRepository().removeModel( domainId, logicalModelRep.getId() );
+      }
+
+      // get updated domain
+      domain = getMetadataDomainRepository().getDomain( domainId );
+
+      if ( domain.getLogicalModels().isEmpty() ) {
```
Comment (mdamour1976) [inline src/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/DSWDatasourceServiceImpl.java]: @IvanNikolaychuk is it possible for domain to be null here and throw NPE?

**Comment by IvanNikolaychuk** [comment]
@mdamour1976, thats a good remar, thanks.
Yes, theoretically, another thread can delete the domain, between the moment we got it, and the moment when we want to get this domain updated.

Null check added now.

**Pull request merged**